### DISCUSSION TITLE
Difficulty pushing image to remote docker repo

### DISCUSSION THREAD

**Discussion by jspruance** [buildpacks/community#198] [Q&A] [answered]
Hello! I am attempting to use the buildpacks Tekton task to clone a Github repo, build the image and then push it to my public repo on Docker Hub. Everything seems to be working until the last step.

**The error info in the pod logs isn't super helpful:**
State:          Terminated
      Reason:       Error
      Message:      [{"key":"StartedAt","value":"2022-11-23T16:49:29.261Z","type":3}]
      Exit Code:    1
      Started:      Wed, 23 Nov 2022 08:49:25 -0800
      Finished:     Wed, 23 Nov 2022 08:49:35 -0800
      
One thing I am questioning is whether I am setting the Docker Hub repo url correctly. After reading the documentation I believe that is what the 'APP_IMAGE' var should reflect. Is that correct?

**From docs:**
APP_IMAGE: The name of where to store the app image. (REQUIRED)
  
  My Docker Hub repo:
  https://hub.docker.com/repository/docker/jspruancedome/ocean-api
  
  ^^ This is the same value I am setting for 'APP_IMAGE'.
  
  The repo doesn't currently contain any images - the one that buildpacks is pushing would be the first image.
  
  Does all of this sound correct? I am pretty new to Tekton, Buildpacks and Kubernetes and trying to ramp up on all of these things but may have a few gaps in my knowledge still. Thanks for any help.

**Comment by AidanDelaney** [comment]
Hi @jspruance and welcome to the Buildpacks community.  The docs for our Tekton task describe https://buildpacks.io/docs/tools/tekton/#43-pipeline as the pipeline definition.  Which takes an input `image` name and passes it as `APP_IMAGE` to the Buildpacks task.  The PipleineRun passes the image name as a parameter:

```yaml
params:
    - name: image
      value: <REGISTRY/IMAGE NAME, eg gcr.io/test/image > 
```

Could you share your `Pipeline` and `PipelineRun` definitions?

**Reply by jspruance** [reply]
Thanks for the reply.

I created a git repo for my working poc...

My PipelineRun definition is in the 'triggers.yaml' file here:
https://github.com/jspruance/build-pipeline-poc/blob/main/docs/getting-started/triggers.yaml

Pipeline definition is here:
https://github.com/tektoncd/triggers/blob/main/docs/getting-started/pipeline.yaml

I believe I have the image name passed in as you described but not sure if I have the format correct (was unsure if it should literally be the name or the repo url):

`params:
          - name: image
            value: hub.docker.com/repository/docker/jspruancedome/ocean-api`

**Comment by jspruance** [comment]
Just to add - it is failing on the 'create' step. Is there any way to get more detailed error messaging so I can see what went wrong?

```
step-create:
    Container ID:  docker://736325f2d2f7fb2cea4f2c355737dba2e528d9e76947361c55a24eb0b32b8606
    Image:         paketobuildpacks/builder:base
    Image ID:      docker-pullable://paketobuildpacks/builder@sha256:dbc641980f88a7f1f8936302d099bbc45e50d35770b00a9379c5e66e34e7d18a
    Port:          <none>
    Host Port:     <none>
    Command:
      /tekton/bin/entrypoint
    Args:
      -wait_file
      /tekton/run/0/out
      -post_file
      /tekton/run/1/out
      -termination_path
      /tekton/termination
      -step_metadata_dir
      /tekton/run/1/status
      -results
      APP_IMAGE_DIGEST
      -entrypoint
      /cnb/lifecycle/creator
      --
      -app=/workspace/source/
      -cache-dir=
      -cache-image=
      -uid=1000
      -gid=1000
      -layers=/layers
      -platform=/platform
      -report=/layers/report.toml
      -process-type=web
      -skip-restore=false
      -previous-image=hub.docker.com/repository/docker/jspruancedome/ocean-api
      -run-image=
      hub.docker.com/repository/docker/jspruancedome/ocean-api
    State:          Terminated
      Reason:       Error
      Message:      [{"key":"StartedAt","value":"2022-11-28T01:01:54.001Z","type":3}]
      Exit Code:    1
      Started:      Sun, 27 Nov 2022 17:01:49 -0800
      Finished:     Sun, 27 Nov 2022 17:01:57 -0800
    Ready:          False
    Restart Count:  0
    Environment:
      CNB_PLATFORM_API:  0.4
    Mounts:
      /layers from layers-dir (rw)
      /platform from empty-dir (rw)
      /tekton/bin from tekton-internal-bin (ro)
      /tekton/creds from tekton-creds-init-home-1 (rw)
      /tekton/home from tekton-internal-home (rw)
      /tekton/results from tekton-internal-results (rw)
      /tekton/run/0 from tekton-internal-run-0 (ro)
      /tekton/run/1 from tekton-internal-run-1 (rw)
      /tekton/run/2 from tekton-internal-run-2 (ro)
      /tekton/steps from tekton-internal-steps (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-xvvsp (ro)
      /workspace from tekton-internal-workspace (rw)
      /workspace/source from ws-4qkrh (rw,path="source")
```

**Comment by jspruance** [comment, marked as answer]
Update: I got this working after changing the 'image' value to just the image / tag instead of the URL of the image:

```
params:
          - name: image
            value: jspruancedome/ocean-api:dome
```

I think the documentation around this is a bit confusing as the example makes it seem like this should be a URL not an image name:

```
params:
    - name: image
      value: <REGISTRY/IMAGE NAME, eg gcr.io/test/image >
      
```

Thanks for the help.
